modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,262 @@
+                state[client][:user] = user_login.name
+                state[client][:pass] = user_login.authentication
+                state[client][:domain] = nil
+                result_code = Net::LDAP::ResultCodeSuccess
+              elsif !user_login.name.empty?
+                if user_login.name =~ /@/
+                  pub_info = user_login.name.split('@')
+                  if pub_info.length <= 2
+                    state[client][:user] = pub_info[0]
+                    state[client][:domain] = pub_info[1]
+                  else
+                    result_code = Net::LDAP::ResultCodeInvalidCredentials
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                  end
+                elsif user_login.name =~ /,/
+                  begin
+                    dn = Net::LDAP::DN.new(user_login.name)
+                    dn.each_pair do |key, value|
+                      if key == 'cn'
+                        names << value
+                      elsif key == 'dc'
+                        domains << value
+                      end
+                    end
+                    state[client][:user] = names.first
+                    state[client][:domain] = domains.empty? ? nil : domains.join('.')
+                  rescue StandardError => e
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                    raise e
+                  end
+                elsif user_login.name =~ /\\/
+                  pub_info = user_login.name.split('\\')
+                  if pub_info.length <= 2
+                    state[client][:user] = pub_info[1]
+                    state[client][:domain] = pub_info[0]
+                  else
+                    result_code = Net::LDAP::ResultCodeInvalidCredentials
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                  end
+                else
+                  state[client][:user] = user_login.name
+                  state[client][:domain] = nil
+                  result_code = Net::LDAP::ResultCodeInvalidCredentials
+                end
+              else
+                state[client][:user] = ''
+                state[client][:domain] = nil
+              end
+
+              state[client][:pass] = user_login.authentication
+
+              unless state[client][:user].empty? && state[client][:pass].empty?
+                report_cred(
+                  ip: state[client][:ip],
+                  port: client.localport,
+                  service_name: 'ldap',
+                  user: state[client][:user],
+                  password: state[client][:pass],
+                  domain: state[client][:domain]
+                )
+              end
+              print_good("LDAP Login Attempt => From:#{state[client][:name]} Username:#{state[client][:user]} Password:#{state[client][:pass]}")
```

Comment:
A  very minor additional detail, if a `domain` was parsed out from the request it would be good to have that in the console output as well.
```suggestion
              result_message = "LDAP Login Attempt => From:#{state[client][:name]} Username:#{state[client][:user]} Password:#{state[client][:pass]}"
              result_message += " Domain: #{state[client][:domain]" if state[client][:domain]
              print_good(result_message)
```